tests/integration/test_tls.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+from pathlib import Path
+from types import SimpleNamespace
+
+import pytest
+import yaml
+from helpers import get_workload_file, oci_image
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+mc = SimpleNamespace(name="mc")
+
+mimir_app_name = "coordinator"
+ca_app_name = "ca"
+app_names = [mimir_app_name, ca_app_name]
+
+
+def get_nginx_config(ops_test: OpsTest):
+    return get_workload_file(
+        ops_test.model_name, mimir_app_name, 0, "nginx", "/etc/nginx/nginx.conf"
+    )
+
+
+@pytest.mark.abort_on_fail
+async def test_nginx_config_has_ssl(ops_test: OpsTest):
```

Comment:
Should we inline and deploy the same bundle you proposed in the PR description?

Reply:
I would add that itest once we have a minimal version of the worker on charmhub so we can avoid the build process of that charm.